pkg/volumes/openstack/volumes.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	cinderv2 "github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/volumeattach"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"kope.io/etcd-manager/pkg/volumes"
+)
+
+const MetadataLatest string = "http://169.254.169.254/openstack/latest/meta_data.json"
+
+type InstanceMetadata struct {
+	Name             string `json:"name"`
+	ProjectID        string `json:"project_id"`
+	AvailabilityZone string `json:"availability_zone"`
```

Comment:
There is no way for me to tell that I am allowing disks to be allowed to mount across zones.  Compute availability zones and Volume availability zones do not need to be 1-to-1.  In most cases these will probably be `nova` for both.  I've chosen to simply ignore AZ for now.

Reply:
does this support situation that compute zone name is like `foobar` and volume az `nova` does it work? We have this situation in all openstacks

Reply:
Yes.  Availability Zone is currently ignored, so this should work.  

I've left it in as a member however I'm not really a huge fan of this style of zoning and am really unsure how to use availability zones to support every Openstack deployment out there.